### ISSUE TITLE
Cannot use `reflex-dom-contrib` package

### ISSUE THREAD

**Issue by tshinohara** [obsidiansystems/obelisk#54]
Hi, thanks for the cool project!

I'm trying to use `reflex-dom-contrib` library in the frontend, but after adding dependencies to `frontend/frontend.cabal` and `ob run`, buiding of `reflex-dom-contrib` fails. Looking at the errors, it seems like a dependencies error, but my nix knowledge is not good enough to figure out whats wrong. Is it not enable to use `reflex-dom-contrib` library?

`frontend/frontend.cabal`
```
library
  hs-source-dirs: src
  build-depends: base
               , common
               , reflex-dom
               , reflex-dom-contrib
               , static
               , text
```
Compile error log.
```
unpacking source archive /nix/store/1i0nd6sc8i3db52mgmjbi6gp6g8r3mid-reflex-dom-contrib-0.4.1.tar.gz
source root is reflex-dom-contrib-0.4.1
/nix/store/j1ly0zzpnzb8y8ry6k4vi4prk8vwq233-set-source-date-epoch-to-latest.sh: line 7: warning: command substitution: ignored null byte in input
setting SOURCE_DATE_EPOCH to timestamp 1455786364 of file reflex-dom-contrib-0.4.1/src/Reflex/Dom/Contrib/Xhr.hs
patching sources
compileBuildDriverPhase
setupCompileFlags: -package-db=/tmp/nix-build-reflex-dom-contrib-0.4.1.drv-0/package.conf.d -j4 -threaded
[1 of 1] Compiling Main             ( Setup.hs, /tmp/nix-build-reflex-dom-contrib-0.4.1.drv-0/Main.o )
Linking Setup ...
configuring
configureFlags: --verbose --prefix=/nix/store/17i361rvpxck5livxbq86il9a0ljjb97-reflex-dom-contrib-0.4.1 --libdir=$prefix/lib/$compiler --libsubdir=$pkgid --docdir=/nix/store/h1yh1nxqf0qksy7934hgr4x5w0lyhl9i-reflex-dom-contrib-0.4.1-doc/share/doc --with-gcc=gcc --package-db=/tmp/nix-build-reflex-dom-contrib-0.4.1.drv-0/package.conf.d --ghc-option=-optl=-Wl,-rpath=/nix/store/17i361rvpxck5livxbq86il9a0ljjb97-reflex-dom-contrib-0.4.1/lib/ghc-8.0.2/reflex-dom-contrib-0.4.1 --ghc-option=-j4 --disable-split-objs --disable-library-profiling --disable-profiling --enable-shared --disable-coverage --enable-library-vanilla --enable-executable-dynamic --enable-tests --ghc-option=-split-sections --ghc-options=-fexpose-all-unfoldings --extra-lib-dirs=/nix/store/sldddw66nndddmjf6knlla50ls5dkkxf-ncurses-6.0-20170902/lib --extra-lib-dirs=/nix/store/h9g62cvpviyhfp3v5k6qwb7k2pafg23a-gmp-6.1.2/lib --extra-include-dirs=/nix/store/hfz0613rjr8xi7pjh4p726rc2hqxf063-gobject-introspection-1.52.1-dev/include --extra-lib-dirs=/nix/store/hfz0613rjr8xi7pjh4p726rc2hqxf063-gobject-introspection-1.52.1-dev/lib --extra-include-dirs=/nix/store/pkizaxn4gf1syfcxnvwrhgadavyw44i8-libffi-3.2.1-dev/include --extra-lib-dirs=/nix/store/pkizaxn4gf1syfcxnvwrhgadavyw44i8-libffi-3.2.1-dev/lib --extra-lib-dirs=/nix/store/8wrrm4r593y99wbdzfjzn7jq1xwc28gl-libffi-3.2.1/lib --extra-include-dirs=/nix/store/4n65jqwiq1gfvwv722pqc2iki3hch7if-glib-2.52.3-dev/include --extra-lib-dirs=/nix/store/4n65jqwiq1gfvwv722pqc2iki3hch7if-glib-2.52.3-dev/lib --extra-include-dirs=/nix/store/h28hdpi9gib4v6f456f5s4281pw22glf-zlib-1.2.11-dev/include --extra-lib-dirs=/nix/store/h28hdpi9gib4v6f456f5s4281pw22glf-zlib-1.2.11-dev/lib --extra-lib-dirs=/nix/store/qjsk4rzv57c3jpdqw5dz36rv2bmxmm4x-zlib-1.2.11/lib --extra-include-dirs=/nix/store/bjhzi3g63ykjlf2zsxm978jkkj5hz0i3-glibc-iconv-2.25-49/include --extra-lib-dirs=/nix/store/swfsmkyb5ls0r8z05q5akq18aafjynw5-glib-2.52.3/lib --extra-lib-dirs=/nix/store/krivj6rixcs7kadxjlyh566kmvh5q0xg-gobject-introspection-1.52.1/lib
Configuring reflex-dom-contrib-0.4.1...
Setup: Encountered missing dependencies:
aeson >=0.8 && <0.11,
base >=4.6 && <4.9,
bifunctors >=4.0 && <5.3,
data-default ==0.5.*,
ghcjs-dom ==0.2.*,
lens >=4.9 && <4.14,
reflex >=0.2 && <0.5,
reflex-dom >=0.2 && <0.4
builder for '/nix/store/7kc29n45hgzrr7zin86cmg0bzl1969lv-reflex-dom-contrib-0.4.1.drv' failed with exit code 1
cannot build derivation '/nix/store/cyp5ndy4r6kj5vscsfkdadzilhnw8m2q-ghc-8.0.2-with-packages.drv': 1 dependencies couldn't be built
error: build of '/nix/store/cyp5ndy4r6kj5vscsfkdadzilhnw8m2q-ghc-8.0.2-with-packages.drv' failed
```

**Comment by srid** [comment]
If a package is not available, you can specify an override for it in your project's `default.nix`. [Here's an example](https://github.com/srid/Taut/blob/master/default.nix#L11-L23) for specifying the `beam` package.

We just need to document how to do this so users can discover it easily.

**Comment by srid** [comment]
As for getting `reflex-dom-contrib` working in particular you might be interesting in [these overrides](https://github.com/srid/deux/blob/2d43b09/default.nix#L27).

**Comment by tshinohara** [comment]
Thank you! Now able to compile.

**Comment by ali-abrar** [comment]
Resolved and documented via #62
